src/component/EditForm.jsx

```diff
@@ -0,0 +1,200 @@
+            boxShadow: 10,
+            background: "white",
+          }}
+        >
+          <Typography
+            sx={{
+              fontWeight: "bold",
+              fontStyle: "italic",
+              fontSize: "35px",
+              textAlign: "center",
+              textDecoration: "underline",
+            }}
+            variant="h5"
+          >
+            Edit User
+          </Typography>
+          {userData && (
+            <Formik
+              initialValues={{
+                firstname: userData.firstname,
+                address: userData.address,
+                phoneno: userData.phoneno,
+                lastname: userData.lastname,
+                email: userData.email,
+              }}
+              validationSchema={Yup.object({
+                firstname: Yup.string().required("Please enter your username"),
+                lastname: Yup.string().required("Please enter your username"),
+                email: Yup.string().email().required("Please enter your email"),
+              })}
+              onSubmit={handleSubmit}
+            >
+              <Form>
+                <Box sx={{ margin: "20px 0" }}>
+                  <Field
+                    as={TextField}
+                    fullWidth
+                    label="firstname"
+                    type="text"
+                    name="firstname"
+                  />
+                </Box>
+                <Box sx={{ margin: "20px 0" }}>
+                  <Field
+                    as={TextField}
+                    fullWidth
+                    label="lastname"
+                    type="text"
+                    name="lastname"
+                  />
+                </Box>
+                <Box sx={{ margin: "20px 0" }}>
+                  <Field
+                    as={TextField}
+                    fullWidth
+                    label="Email"
+                    type="email"
+                    name="email"
+                  />
+                </Box>{" "}
+                <Box sx={{ margin: "20px 0" }}>
+                  <Field
+                    as={TextField}
+                    fullWidth
+                    label="Addres"
+                    type="text"
+                    name="address"
+                  />
+                </Box>{" "}
+                <Box sx={{ margin: "20px 0" }}>
+                  <Field
+                    as={TextField}
+                    fullWidth
+                    label="phoneno"
+                    type="text"
+                    name="phoneno"
+                  />
+                </Box>
+                <Button
+                  sx={{
+                    background:
+                      "linear-gradient(90.9deg, rgb(3, 195, 195) 0.3%, rgb(37, 84, 112) 87.8%)",
+                  }}
+                  variant="contained"
+                  type="submit"
+                >
+                  Save Changes
+                </Button>
+                <NavLink to={"/home"}>
```

Comment:
dont make this / home route , use/ for both login and dashboard, 
make it conditional rendering